### PR TITLE
Kotlin Conversion: SearchResults

### PR DESCRIPTION
**Pull request by sharvaniharan**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package org.wikipedia.search
+
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.mwapi.MwQueryPage
+
+class SearchResults {
+    var results: List<SearchResult>
+    var continuation: Map<String, String>?
+    var suggestion: String? = null
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:9]: This can become a `data class` with these fields in the primary constructor.

**Review by dbrant — CHANGES_REQUESTED** [review]

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        this.results = results
+        continuation = null
+    }
+
+    constructor(pages: List<MwQueryPage?>, wiki: WikiSite,
+                continuation: Map<String, String>?, suggestion: String?) {
+        val searchResults: MutableList<SearchResult> = ArrayList()
+
+        // Sort the array based on the "index" property
+        (pages as ArrayList).sortWith { o2, o1 -> (o1?.index()!!.compareTo(o2?.index()!!)) }
```
[the remark this answers is not in the document]
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/search/SearchResults.kt]: Thank you Dmitry!! 🤦‍♀️

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package org.wikipedia.search
+
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.mwapi.MwQueryPage
+
+data class SearchResults(
+        var results: List<SearchResult>,
+        var continuation: Map<String, String>?,
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:8]: Would it be possible to change `List` to `MutableList`?

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        var suggestion: String? = null) {
+
+    constructor() : this(ArrayList(), null)
+
+    constructor(results: List<SearchResult>) : this(results, null)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:13]: Same

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    constructor(pages: List<MwQueryPage>, wiki: WikiSite,
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:15]: Same.

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+                continuation: Map<String, String>?, suggestion: String?) : this() {
+        val searchResults: MutableList<SearchResult> = ArrayList()
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:17]: If the list is `Mutable`, then we don't need to create this.

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+        // Sort the array based on the "index" property
+        (pages as ArrayList).sortWith { o1, o2 -> (o1?.index()!!.compareTo(o2?.index()!!)) }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:20]: The `as ArrayList` can be removed.

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+        for (page in pages) {
+            searchResults.add(SearchResult(page, wiki))
+        }
+
+        results = searchResults
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:26]: If the list is `Mutable`, then we don't need to create this.

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/search/SearchResults.kt [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        var suggestion: String? = null) {
+
+    constructor() : this(ArrayList(), null)
+
+    constructor(results: MutableList<SearchResult>) : this(results, null)
+
+    constructor(pages: MutableList<MwQueryPage>, wiki: WikiSite,
+                continuation: Map<String, String>?, suggestion: String?) : this() {
+        // Sort the array based on the "index" property
+        (pages).sortWith { o1, o2 -> (o1?.index()!!.compareTo(o2?.index()!!)) }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/search/SearchResults.kt:18]: Parentheses can be removed

**Pull request merged**